client/ayon_silhouette/plugins/publish/extract_workfile.py

```diff
@@ -0,0 +1,39 @@
+import os
+
+from ayon_core.pipeline import (
+    publish,
+    registered_host
+)
+from ayon_silhouette.api import lib
+
+
+class SilhouetteExtractWorkfile(publish.Extractor):
+    label = "Extract Workfile"
+    hosts = ["silhouette"]
+    families = ["workfile"]
+
+    def process(self, instance):
+        """Extract the current working file as .zip"""
+        # Note that Silhouette project workfiles are actually folders,
+        # not files.
+
+        current_file = instance.context.data["currentFile"]
+        if not current_file:
+            raise publish.PublishError("No current file found in context")
+
+        # Save current file
+        host = registered_host()
+        host.save_workfile()
+
+        # Zip current workfile (Silhouette workfiles are folders)
+        staging_dir = self.staging_dir(instance)
+        filename = f"{instance.name}.zip"
+        lib.zip_folder(current_file, os.path.join(staging_dir, filename))
+
+        # Add representation
+        instance.data.setdefault("representations", []).append({
+            "name": "zip",
```

Comment:
Maybe using more specific name would help?
```suggestion
            "name": "workfile",
```

Reply:
Help with what specifically? I don't have any strong opinion about either.

Reply:
Seems like Harmony integration (which also zips the files) uses `tpl` as name. Which I suppose is harmony's workfile extension: https://github.com/ynput/ayon-harmony/blob/5a93fb2993d068cf2bfc95438c5eab55e4eba35b/client/ayon_harmony/plugins/publish/extract_workfile.py#L37-L39

So I guess, if we were to do the equivalent the name here would be `sfx` - but it may be even more confusing? :D

Reply:
From brief private chat with @iLLiCiTiT we went for `sfx_zip`. @antirotor @jakubjezek001 let me know your thoughts.

The commit is here: https://github.com/ynput/ayon-silhouette/pull/14/commits/9ec84ad258d698cec94d04cf85c09e0c6a42dfa9